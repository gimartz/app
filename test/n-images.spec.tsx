import * as React from 'react';
import { render } from '@testing-library/react';
import { ImageListType } from '../src';
import App from './fixtures/app';

describe('testing initial list of n images', () => {
  it('should render component', () => {
    const file = new File([''], 'image-1-name', { type: 'image/png' });

    const images: ImageListType = [
      {
        file,
        dataURL: 'image 1',
      },
    ];
    const { container } = render(<App value={images} />);
    expect(container).toMatchSnapshot();
  });
});
