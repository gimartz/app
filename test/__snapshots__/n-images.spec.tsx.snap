// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing initial list of n images should render component 1`] = `
<div>
  <div
    id="app"
  >
    <input
      accept="image/*"
      style="display: none;"
      type="file"
    />
    <div
      class="upload__image-wrapper"
    >
      <button
        id="btn-upload"
        type="button"
      >
        Click or Drop here
      </button>
      <button
        id="btn-remove"
        type="button"
      >
        Remove all images
      </button>
      <div
        id="list-images"
      >
        <div
          class="image-item"
        >
          <img
            alt="preview"
            id="image_0"
            src="image 1"
          />
          <div
            class="image-item__btn-wrapper"
          >
            <button
              id="update_0"
              type="button"
            >
              Update 0
            </button>
            <button
              id="remove_0"
              type="button"
            >
              Remove 0
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
